### ISSUE TITLE
Uncaught ReferenceError: JQPlotChartFactory is not defined

### ISSUE THREAD

**Issue by liviuconcioiu** [phpmyadmin/phpmyadmin#17420]
### Describe the bug

I just run a query that took 2 seconds then I clicked on profiling.

### To Reproduce

1. Create a database (`test`) with 15000 tables
2. Run this query
```sql
SELECT *, `TABLE_SCHEMA` AS `Db`, `TABLE_NAME` AS `Name`, `TABLE_TYPE` AS `TABLE_TYPE`, `ENGINE` AS `Engine`, `ENGINE` AS `Type`, `VERSION` AS `Version`, `ROW_FORMAT` AS `Row_format`, `TABLE_ROWS` AS `Rows`, `AVG_ROW_LENGTH` AS `Avg_row_length`, `DATA_LENGTH` AS `Data_length`, `MAX_DATA_LENGTH` AS `Max_data_length`, `INDEX_LENGTH` AS `Index_length`, `DATA_FREE` AS `Data_free`, `AUTO_INCREMENT` AS `Auto_increment`, `CREATE_TIME` AS `Create_time`, `UPDATE_TIME` AS `Update_time`, `CHECK_TIME` AS `Check_time`, `TABLE_COLLATION` AS `Collation`, `CHECKSUM` AS `Checksum`, `CREATE_OPTIONS` AS `Create_options`, `TABLE_COMMENT` AS `Comment` FROM `information_schema`.`TABLES` t WHERE `TABLE_SCHEMA` IN ('test') ORDER BY Name ASC LIMIT 250 OFFSET 1750;
```
3. Click on `Profiling`

### Expected behavior

The chart pie should appear and no error should be given.

### Screenshots

https://user-images.githubusercontent.com/25424343/173256407-59ebea09-568a-41d3-9c43-2ee800ad045d.mp4

### Server configuration

- Operating system: CentOS Stream 9
- Web server: nginx/1.22
- Database version: 10.5.13-MariaDB-log - MariaDB Server
- PHP version: 8.1.7
- phpMyAdmin version: 5.3.0-dev+20220611.3e6b0abbe2

### Client configuration

- Browser: Firefox/102
- Operating system: Windows 10

### Additional context

```JSON
{
    "pma_version": "5.3.0-dev+20220611.3e6b0abbe2",
    "browser_name": "FIREFOX",
    "browser_version": "102.0",
    "user_os": "Win",
    "server_software": "nginx/1.22.0",
    "user_agent_string": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:102.0) Gecko/20100101 Firefox/102.0",
    "locale": "en",
    "configuration_storage": "enabled",
    "php_version": "8.1.7",
    "script_name": "index.php",
    "exception_type": "js",
    "exception": {
        "mode": "stack",
        "name": "ReferenceError",
        "message": "JQPlotChartFactory is not defined",
        "stack": [
            {
                "func": "Functions.createProfilingChart",
                "line": "2270",
                "column": "17",
                "context": [
                    " */",
                    "",
                    "",
                    "Functions.createProfilingChart = function (target, data) {",
                    "  // create the chart",
                    "  var factory = new JQPlotChartFactory();",
                    "  var chart = factory.createChart(ChartType.PIE, target); // create the dat//...",
                    "",
                    "  var dataTable = new DataTable();",
                    "  dataTable.addColumn(ColumnType.STRING, '');",
                    "  dataTable.addColumn(ColumnType.NUMBER, '');"
                ],
                "uri": "js/dist/functions.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/dist/functions.js"
            },
            {
                "func": "Sql.makeProfilingChart",
                "line": "1043",
                "column": "13",
                "context": [
                    "    data.push([key, parseFloat(value)]);",
                    "  }); // Remove chart and data divs contents",
                    "",
                    "  $('#profilingchart').html('').show();",
                    "  $('#profilingChartData').html('');",
                    "  Functions.createProfilingChart('profilingchart', data);",
                    "};",
                    "/**",
                    " * initialize profiling data tables",
                    " */",
                    ""
                ],
                "uri": "js/dist/sql.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/dist/sql.js"
            },
            {
                "func": "?",
                "line": "1084",
                "column": "7",
                "context": [
                    "    }",
                    "  });",
                    "};",
                    "",
                    "AJAX.registerOnload('sql.js', function () {",
                    "  Sql.makeProfilingChart();",
                    "  Sql.initProfilingTables();",
                    "});",
                    "",
                    "//# sourceMappingURL=sql.js.map"
                ],
                "uri": "js/dist/sql.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/dist/sql.js"
            },
            {
                "func": "newFunc",
                "line": "258",
                "column": "23",
                "context": [
                    "   */",
                    "  wrapFunction: function (func) {",
                    "    if (!func.wrapped) {",
                    "      var newFunc = function () {",
                    "        try {",
                    "          return func.apply(this, arguments);",
                    "        } catch (x) {",
                    "          TraceKit.report(x);",
                    "        }",
                    "      };",
                    ""
                ],
                "uri": "js/dist/error_report.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/dist/error_report.js"
            },
            {
                "func": "dispatch",
                "line": "2",
                "column": "43064",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "add/v.handle",
                "line": "2",
                "column": "41048",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "trigger",
                "line": "2",
                "column": "71515",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "trigger/<",
                "line": "2",
                "column": "72110",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "each",
                "line": "2",
                "column": "3003",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "each",
                "line": "2",
                "column": "1481",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "trigger",
                "line": "2",
                "column": "72086",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "fireOnload",
                "line": "130",
                "column": "17",
                "context": [
                    "   *",
                    "   * @return {void}",
                    "   */",
                    "  fireOnload: function (file) {",
                    "    var eventName = 'onload_' + AJAX.hash(file);",
                    "    $(document).trigger(eventName);",
                    "",
                    "    if (this.debug) {",
                    "      // eslint-disable-next-line no-console",
                    "      console.log( // no need to translate",
                    "      'Fired event ' + eventName + ' for file ' + file);"
                ],
                "uri": "js/dist/ajax.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/dist/ajax.js"
            },
            {
                "func": "done",
                "line": "773",
                "column": "14",
                "context": [
                    "     *",
                    "     * @return {void}",
                    "     */",
                    "    done: function (script, callback) {",
                    "      if ($.inArray(script, this.scriptsToBeFired)) {",
                    "        AJAX.fireOnload(script);",
                    "      }",
                    "",
                    "      if ($.inArray(script, this.scriptsToBeLoaded)) {",
                    "        this.scriptsToBeLoaded.splice($.inArray(script, this.scriptsToBeLoa//...",
                    "      }"
                ],
                "uri": "js/dist/ajax.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/dist/ajax.js"
            },
            {
                "func": "load",
                "line": "755",
                "column": "16",
                "context": [
                    "",
                    "        if ($.inArray(script, self.scripts) === -1) {",
                    "          this.add(script);",
                    "          this.appendScript(script, callback);",
                    "        } else {",
                    "          self.done(script, callback);",
                    "        }",
                    "      } // Trigger callback if there is nothing else to load",
                    "",
                    "",
                    "      self.done(null, callback);"
                ],
                "uri": "js/dist/ajax.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/dist/ajax.js"
            },
            {
                "func": "responseHandler/<",
                "line": "578",
                "column": "30",
                "context": [
                    "        if (data.params) {",
                    "          CommonParams.setAll(data.params);",
                    "        }",
                    "",
                    "        if (data.scripts) {",
                    "          AJAX.scriptHandler.load(data.scripts);",
                    "        }",
                    "",
                    "        if (data.displayMessage) {",
                    "          $('#page_content').prepend(data.displayMessage);",
                    "          Functions.highlightSql($('#page_content'));"
                ],
                "uri": "js/dist/ajax.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/dist/ajax.js"
            },
            {
                "func": "reset",
                "line": "839",
                "column": "7",
                "context": [
                    "       * on pages and submissions of forms",
                    "       */",
                    "",
                    "      $(document).off('click', 'a').on('click', 'a', AJAX.requestHandler);",
                    "      $(document).off('submit', 'form').on('submit', 'form', AJAX.requestHa//...",
                    "      callback();",
                    "    }",
                    "  }",
                    "};",
                    "/**",
                    " * Here we register a function that will remove the onsubmit event from all"
                ],
                "uri": "js/dist/ajax.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/dist/ajax.js"
            },
            {
                "func": "responseHandler",
                "line": "522",
                "column": "26",
                "context": [
                    "        AJAX.active = false;",
                    "        AJAX.xhr = null;",
                    "        return;",
                    "      }",
                    "",
                    "      AJAX.scriptHandler.reset(function () {",
                    "        if (data.reloadNavigation) {",
                    "          Navigation.reload();",
                    "        }",
                    "",
                    "        if (data.title) {"
                ],
                "uri": "js/dist/ajax.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/dist/ajax.js"
            },
            {
                "func": "c",
                "line": "2",
                "column": "28327",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "fireWith",
                "line": "2",
                "column": "29072",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "l",
                "line": "2",
                "column": "79901",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "send/o/<",
                "line": "2",
                "column": "82355",
                "context": [
                    "/*! jQuery v3.6.0 | (c) OpenJS Foundation and other contributors | jquery.o//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.3.0-dev%2B20220611.3e6b0abbe2",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            }
        ],
        "incomplete": "false",
        "uri": "index.php?route=%2Fdatabase%2Fsql"
    }
}
```

**Comment by liviuconcioiu** [comment]
@williamdes I have updated the bug report on how to reproduce the issue and also a video showing the bug. On QA_5_2 it doesn't show the pie either, but no JS error is given. I think it depends on query execution time, so this should be fixed on QA_5_2.

**Comment by williamdes** [comment]
> @williamdes I have updated the bug report on how to reproduce the issue and also a video showing the bug. On QA_5_2 it doesn't show the pie either, but no JS error is given. I think it depends on query execution time, so this should be fixed on QA_5_2.

Thank you, this will be very helpful

**Comment by williamdes** [comment]
Currently this does not throw an error, but the chart does not render, if you hit "Refresh" below the query it will show up

**Comment by MauricioFauth** [comment]
Fixed by 5f7499abb0e7861b7044303d0d0cb4f41f40a69f.
